README.md

```diff
@@ -42,9 +42,38 @@ resource "docker_image" "nginx" {
 }
 
 # Create a docker container resource
-# -> same as 'docker run --name nginx -d nginx:latest'
+# -> same as 'docker run --name nginx -p8080:80 -d nginx:latest'
 resource "docker_container" "nginx" {
   name    = "nginx"
   image   = docker_image.nginx.latest
+
+  ports {
+    external = 8080
+    internal = 80
+  }
+}
+
+# Or create a service resource
+# -> same as 'docker service create -d -p 8081:80 --name nginx-service --replicas 2 nginx:latest'
+resource "docker_service" "nginx_service" {
+  name = "nginx-service"
+  task_spec {
+    container_spec {
+      image = docker_image.nginx.latest
```

Comment:
Deprecated `latest` attribute.  Based on what I'm seeing in https://github.com/kreuzwerker/terraform-provider-docker/blob/fix/service-image-name/internal/provider/data_source_docker_image.go, a simple `docker_image.nginx.id` should work the same way, and we don't need the new `sha256_digest` there, either.

How does using the local image `id` or registry image `sha256_digest` work versus using a `<registry>/<image>:<tag>[@<sha256>]` hash, when it comes to a docker service, if the task is scheduled on a node that the provider isn't currently communicating with (where, presumably the `docker_image` or `docker_registry_image` would be located)?

I think it's the local node's docker engine that does the pull from the registry if it's not locally available.  Having it try to pull a hash probably doesn't give it all the information it needs...
```
m1:~$ docker image inspect 075d847609aa
[
    {
        "Id": "sha256:075d847609aa84dbbe8a5a36ca3852dd75bbcd643a92bb5e1757e75727d44210",
        "RepoTags": [
            "jakesys/aws2:latest"
        ],
        "RepoDigests": [
            "jakesys/aws2@sha256:aa579322e9a0a27c5bece3cc9f5a76e5482c2c846e555917f17ce28034a78d3a"
        ],
... 
]
```
Perhaps what we really want to provide here is the image's `RepoDigest` string.

Reply:
> a simple docker_image.nginx.id should work the same way, and we don't need the new sha256_digest there, either.

Yes, that's true.

> Having it try to pull a hash probably doesn't give it all the information it needs...

Good point. I did not think of all the swarm node. So either 
- `<registry>/<image>:<tag>` or
- `<registry>/<image>@<sha256>` or
- `<registry>/<image>:<tag>@<sha256>`
are the ways to go

Reply:
While tracking down what's in `RepoDigests`, I discovered that this is _empty_ for locally-built images that have not been pushed to a registry -- IIRC, @suzuki-shunsuke already alluded to this.

So, for references to local `docker_image` data sources and references, the use of `id` should be used, too.

Reply:
See 18c3f355c27c5a82480f3eb9a5053144adcc7f4f

Reply:
> So, for references to local docker_image data sources and references, the use ofid should be allowed, too.

and `repo_digest` as well